files/class-vip-filesystem.php

```diff
@@ -197,6 +202,24 @@ public function filter_validate_file( $file ) {
 		return $file;
 	}
 
+	/**
+	 * Check if file already exists and return a unique filename if it does.
+	 *
+	 * @param  string  Path starting with /wp-content/uploads
+	 */
+	protected function validate_file_unique_name( $file_name, $upload_path ) {
+		$result = $this->stream_wrapper->client->is_file( $upload_path . $file_name );
+		
+		if ( true === $result ) {
+			$file_root = pathinfo( $file_name, PATHINFO_FILENAME );
+			$ext       = pathinfo( $file_name, PATHINFO_EXTENSION );
+			$timestamp = time();
```

Comment:
We may want to use `microtime` here and either create a random suffix based on that (or append an additional randomized string) to avoid the rare edge case collision where two files with the same name are uploaded at the exact same time.

Reply:
Another option I realized is to re-use the `validate_file_type` call. That calls `get_unique_filename` on the API, which returns a unique name if a file already exists with the provided name. That way, we don't need to introduce an additional API call the upload flow.

We'll just want to verify the the API endpoint actually does return a unique name :)

Reply:
I did think about using `get_unique_filename` but when I looked at it in [wpvip-files service](https://github.com/Automattic/wpvip-files-service/blob/1d752be7cf906f2a670516cf8b1b646b8b69685f/src/class-files-service.php#L164), it seemed like it was basically doing the same thing that WP  does.